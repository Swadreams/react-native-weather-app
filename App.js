/**
 * Coupa Weather Project
 **/

import React from 'react';
import 'react-native-gesture-handler';
import {NavigationContainer} from '@react-navigation/native';
import {createStackNavigator} from '@react-navigation/stack';
import {Provider} from 'react-redux';
import configureStore from './src/store/configureStore';

import {LandingPage} from './src/components/screens/LandingPage';
import {HistoricalWeather} from './src/components/screens/HistoricalWeather';
import {WeatherCurrentDetails} from './src/components/screens/WeatherCurrentDetails';

const Stack = createStackNavigator();

const HeaderStyle = {
  headerShown: true,
  headerTransparent: true,
  headerTintColor: '#fff',
  headerTitleStyle: {
    fontWeight: 'bold',
  },
};

const store = configureStore();

const App = () => {
  return (
    <Provider store={store}>
      <NavigationContainer>
        <Stack.Navigator>
          <Stack.Screen
            name="Locations"
            component={LandingPage}
            options={{headerShown: false}}
          />
          <Stack.Screen
            name="Details"
            component={WeatherCurrentDetails}
            options={HeaderStyle}
          />
          <Stack.Screen
            name="Historical"
            component={HistoricalWeather}
            options={HeaderStyle}
          />
        </Stack.Navigator>
      </NavigationContainer>
    </Provider>
  );
};

export default App;
