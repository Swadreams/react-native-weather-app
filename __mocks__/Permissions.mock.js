export const RESULTS = {
  UNAVAILABLE: 'unavailable',
  DENIED: 'denied',
  BLOCKED: 'blocked',
  GRANTED: 'granted',
};

export const openSettings = jest.fn(async () => {});

export const check = jest.fn(() => {
  return Promise.resolve(RESULTS.GRANTED);
});

export const request = jest.fn(() => {
  return Promise.resolve(RESULTS.GRANTED);
});

export const PERMISSIONS = {ANDROID: {}, IOS: {}};
