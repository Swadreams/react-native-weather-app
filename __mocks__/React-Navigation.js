const MockNavigation = {
  replace: jest.fn(),
  push: jest.fn(),
  pop: jest.fn(),
  popToTop: jest.fn(),
  goBack: jest.fn(),
  navigate: jest.fn(),
  reset: jest.fn(),
  setParams: jest.fn(),
  dispatch: jest.fn(),
  isFocused: jest.fn().mockReturnValue(true),
  canGoBack: jest.fn().mockReturnValue(true),
  addListener: jest.fn(),
  removeListener: jest.fn(),
  dangerouslyGetParent: jest.fn(),
  dangerouslyGetState: jest.fn(),
  setOptions: jest.fn(),
};

const ReactNavigationNative = jest.requireActual('@react-navigation/native');

const MockNativeNavigation = {
  ...ReactNavigationNative,
  useLinking: jest.fn(),
  useNavigation: jest.fn().mockReturnValue(MockNavigation),
  useFocusEffect: jest.fn().mockImplementation(callback => {
    const unsubscribe = callback();
    if (typeof unsubscribe === typeof Function) {
      unsubscribe();
    }
    return unsubscribe;
  }),
  useIsFocused: jest.fn().mockReturnValue(true),
  DrawerActions: {
    toggleDrawer: jest.fn(),
  },
};

const {createStackNavigator} = jest.requireActual('@react-navigation/stack');

export const MockStackNavigation = {
  createStackNavigator,
  CardStyleInterpolators: {},
};

export {MockNavigation, MockNativeNavigation};
