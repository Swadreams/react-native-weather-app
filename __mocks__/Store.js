import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const middlewares = [thunk];

const initialState = {
  weather: {
    locations: [],
    currentLocation: {},
    currentWeather: {},
    currentWeatherLocation: {},
    historicalWeather: {},
    homeWeather: {},
    homeLocation: {},
    homeWeatherLocation: {},
    isLoading: {},
  },
};

let store;

export const configureMockStore = data => {
  const initialData = data ? {...initialState, ...data} : initialState;
  store = configureStore(middlewares)(initialData);
  return store;
};

export const getMockStore = () => {
  return store;
};
