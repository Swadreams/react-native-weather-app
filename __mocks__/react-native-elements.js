jest.mock('react-native-elements', () => {
  const SearchBar = {
    placeholder: '',
    onChangeText: jest.fn(),
    value: jest.fn(),
    lightTheme: jest.fn(),
  };

  return {
    ...SearchBar,
  };
});
