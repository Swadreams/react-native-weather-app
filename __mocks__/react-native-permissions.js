import * as MockPermissions from './Permissions.mock';
jest.mock('react-native-permissions', () => MockPermissions);
