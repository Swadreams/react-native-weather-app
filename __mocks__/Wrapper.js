import React from 'react';
import {Provider} from 'react-redux';

import {MockNavigation} from './React-Navigation';
import {getMockStore} from './Store';
import {Locations} from '../src/components/screens/LandingPage';
import {Geolocation} from './@react-native-community/geolocation';
import {PERMISSIONS, request} from './react-native-permissions';

const MockWrapper = ({component, initialRoute}) => {
  const screenName = 'Locations';
  const store = getMockStore();

  const ComponentWithNavProps = () => {
    return (
      <component
        route={{...initialRoute, key: screenName, name: Locations}}
        navigation={MockNavigation}
      />
    );
  };

  return store ? (
    <Provider store={store}>
      <ComponentWithNavProps />
    </Provider>
  ) : (
    <ComponentWithNavProps />
  );
};

MockWrapper.defaultProps = {
  initialRoute: {},
};

export {MockWrapper};
