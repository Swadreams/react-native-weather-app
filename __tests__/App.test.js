/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';

// Note: test renderer must be required after react-native.
import renderer, {create, act} from 'react-test-renderer';
import {MockWrapper} from '../__mocks__/Wrapper';
import {configureMockStore} from '../__mocks__/Store';

// it('renders correctly', () => {
//   renderer.create(<MockWrapper component={App} />);
// });

const userSession = {
  locations: [],
  currentLocation: {},
  currentWeather: {},
  currentWeatherLocation: {},
  historicalWeather: {},
  homeWeather: {},
  homeLocation: {},
  homeWeatherLocation: {},
  isLoading: {},
};

describe('App Component ', () => {
  let component;
  let root;
  let store;

  beforeEach(() => {
    // component = create(<MockWrapper component={App} />);
    // ({root} = component);

    store = configureMockStore({
      userSession,
    });
    store.clearActions();
    component = create(<MockWrapper component={App} />);
    ({root} = component);
  });

  it('renders correctly', async () => {
    let tree;
    await act(async () => {
      jest.runAllTimers();
      tree = await component.toJSON();
    });
    expect(tree).toMatchSnapshot();
  });
});
