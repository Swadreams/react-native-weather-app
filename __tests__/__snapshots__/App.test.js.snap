// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component  renders correctly 1`] = `
<component
  navigation={
    Object {
      "addListener": [MockFunction],
      "canGoBack": [MockFunction],
      "dangerouslyGetParent": [MockFunction],
      "dangerouslyGetState": [MockFunction],
      "dispatch": [MockFunction],
      "goBack": [MockFunction],
      "isFocused": [MockFunction],
      "navigate": [MockFunction],
      "pop": [MockFunction],
      "popToTop": [MockFunction],
      "push": [MockFunction],
      "removeListener": [MockFunction],
      "replace": [MockFunction],
      "reset": [MockFunction],
      "setOptions": [MockFunction],
      "setParams": [MockFunction],
    }
  }
  route={
    Object {
      "key": "Locations",
      "name": undefined,
    }
  }
/>
`;
